### DISCUSSION TITLE
Getting click event of the model as coordinate

### DISCUSSION THREAD

**Discussion by softmarshmallow** [google/model-viewer#2613] [Q&A]
From example, i cannot find example of getting the click event of the model.

The final approach that i want to establish is, getting click event, displaying annotations on to that clicked pos of model.

### References:
- https://modelviewer.dev/examples/annotations/index.html

**Comment by elalish** [comment]
Does this help? https://github.com/google/model-viewer/blob/master/packages/modelviewer.dev/src/tester.ts#L147

**Comment by softmarshmallow** [comment]
> Does this help? https://github.com/google/model-viewer/blob/master/packages/modelviewer.dev/src/tester.ts#L147

Not sure if this will return the value i'm expecting, Will try and let you know.

Thanks for the fast support :)

**Comment by shamsail** [comment]
@softmarshmallow the link from the @elalish should work. As, I was also using this methodology to display the annotations on the desired clicked position when I was working on the annotations on my project.

But keep in mind two to three things. you have to focus on `position` and `normal` separately that you will get from `viewer.positionAndNormalFromPoint(x, y)` for further use (in case you want to automate hotspots).

Another important thing, if you are working with two `model-viewer` instances on the same page than `event.clientX` will not behave correctly. For this, you have to use `event.layerX` etc. See this issue as well https://github.com/google/model-viewer/issues/2554  (I did not use this thing, but it's good to know in case of multiple instances)

If you found your solution feel free to close the issue and open it again if you need any help.
